lib/usd/hdMaya/delegates/sceneDelegate.cpp

```diff
@@ -595,6 +605,16 @@ void HdMayaSceneDelegate::InsertDag(const MDagPath& dag)
 
 void HdMayaSceneDelegate::NodeAdded(const MObject& obj) { _addedNodes.push_back(obj); }
 
+void HdMayaSceneDelegate::NodeRemoved(const MObject& obj)
+{
+    // Note: remove in backward order to avoid processing items twice.
+    for (size_t i = _addedNodes.size() - 1; i < _addedNodes.size(); ++i) {
```

Comment:
This for loop looks like a bug

Reply:
Also, in terms of algorythmic complexity, it,s better to do something like: std::removeif(...); _addedNodes.erase(last, end);

Reply:
The loop looks broken, but actually works because size_t is unsigned. I can use remove_if + erase though, it is slightly more efficient when there are a lot of removal. I don't expect the add + remove notifications without any update in between to happen frequently, only in rare corner cases like when the renderer adds a temp light, but the proposed change is simple.

Reply:
Hmm, I would believe that the loop worked if there was --i and not ++i. With ++i, you always only check the last element

Reply:
Oh you are right! I misinterpreted you comment, I thought you were commenting on the comparison, not the increment that should have been a decrement.  Well, remove_if fixed that, just goes to show to always use the std algorithms....